src/corelibs/U2Formats/src/BAMUtils.cpp

```diff
@@ -198,7 +212,7 @@ void BAMUtils::convertToSamOrBam(const GUrl &samUrl, const GUrl &bamUrl, const C
             samwrite(out, b);    // write the alignment to `out'
         }
 
-        samreadCheck(r, os, sourceName);
+        samreadCheck(r, os, sourceFileUrl.getURLString());
```

Comment:
Specify the template explicitly to avoid misunderstandings. C++11 can't detect a template type every time, but C++17 can and we do not support this standard, so it may cause some problems.

Reply:
done